ucrop/src/main/java/com/yalantis/ucrop/util/BitmapLoadUtils.java

```diff
@@ -50,14 +50,10 @@ public static Bitmap decode(@NonNull Context context, @Nullable Uri uri,
             close(parcelFileDescriptor);
         }
 
-        ExifInterface exif = getExif(uri);
-        if (exif != null) {
-            int exifOrientation = exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL);
-            // TODO Should not rotate bitmap but initially apply needed angle to the matrix
-            return rotateBitmap(decodeSampledBitmap, exifToDegrees(exifOrientation));
-        } else {
-            return decodeSampledBitmap;
-        }
+
+        int exifOrientation = getExifOrientation(context, uri);
+        // TODO Should not rotate bitmap but initially apply needed angle to the matrix
```

Comment:
I think this will be a bit tricky, since the current setup logic depends on the `intrinsicWidth` and `intrinsicHeight`s of the `Drawable`, which will be the original dimensions if we don't create a new bitmap. We'd also need special cases that handle the other orientations (not sure if we can mirror bitmaps in place currently without going into native stuff like JNI).